dvc/repo/checkout.py

```diff
@@ -85,28 +90,71 @@ def checkout(
     if isinstance(targets, str):
         targets = [targets]
 
-    pairs = _collect_pairs(self, targets, with_deps, recursive)
-    total = get_all_files_numbers(pairs)
+    def onerror(target, exc):
+        if target and isinstance(
+            exc,
+            (
+                StageFileDoesNotExistError,
+                StageFileBadNameError,
+                NoOutputOrStageError,
+            ),
+        ):
+            raise CheckoutErrorSuggestGit(target) from exc
+        raise  # pylint: disable=misplaced-bare-raise
+
+    view = self.index.targets_view(
+        targets,
+        recursive=recursive,
+        with_deps=with_deps,
+        onerror=onerror,
+    )
+
+    old = build_data_index(view, self.root_dir, self.fs, compute_hash=True)
+    new = view.data["repo"]
+
+    for out in view.outs:
+        out.changed_cache()
```

Comment:
This is suboptimal. We need to just pass the index, collect oids in it and check all of them (probably parallelized).